util/pkg/vfs/context.go

```diff
@@ -422,3 +430,35 @@ func (c *VFSContext) buildOSSPath(p string) (*OSSPath, error) {
 
 	return NewOSSPath(c.ossClient, bucket, u.Path)
 }
+
+func (c *VFSContext) buildVaultPath(p string) (*VaultPath, error) {
+	u, err := url.Parse(p)
+	if err != nil {
+		return nil, fmt.Errorf("invalid vault url: %q", p)
+	}
+
+	if c.vaultClient == nil {
+
+		if u.Scheme != "vault" {
+			return nil, fmt.Errorf("invalid vault url: %q", p)
+		}
+
+		queryValues := u.Query()
+
+		schema := "https://"
+		if queryValues.Get("tls") == "false" {
+			schema = "http://"
+		}
+
+		vaultClient, err := newVaultClient(schema + u.Host)
+		if err != nil {
+			return nil, err
+		}
+		err = authenticateVaultClient(vaultClient)
```

Comment:
Perhaps `newVaultClient()` could also authenticate? What's the purpose of being able to get an unauthenticated Vault client?

Reply:
Originally, it was in order to do testing. But right now, this can probably be moved in as tests use this factory function anyway.

Reply:
Moved tis in now